AdobeMediaGallery/Test/Unit/Model/Asset/Command/DeleteByIdTest.php

```diff
@@ -0,0 +1,177 @@
+ */
+class DeleteByIdTest extends TestCase
+{
+    /**
+     * @var DeleteById
+     */
+    private $deleteMediaAssetById;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var GetByIdInterface
+     */
+    private $getMediaAssetById;
+
+    /**
+     * @var AssetInterfaceFactory
+     */
+    private $mediaAssetFactory;
+
+    /**
+     * @var AssetInterface
+     */
+    private $mediaAsset;
+
+    /**
+     * @var AdapterInterface
+     */
+    private $adapter;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Initialize basic test class mocks
+     */
+    protected function setUp()
+    {
+        $this->resourceConnection = $this->createMock(ResourceConnection::class);
+        $this->getMediaAssetById = $this->createMock(GetByIdInterface::class);
+        $this->mediaAssetFactory = $this->createMock(AssetInterfaceFactory::class);
+        $this->logger = $this->createMock(LoggerInterface::class);
+
+        $this->deleteMediaAssetById = (new ObjectManager($this))->getObject(
+            DeleteById::class,
+            [
+                'resourceConnection' => $this->resourceConnection,
+                'getMediaAssetById' => $this->getMediaAssetById,
+                'mediaAssetFactory' =>  $this->mediaAssetFactory,
+                'logger' =>  $this->logger,
+            ]
+        );
+
+        $this->mediaAsset = $this->createMock(AssetInterface::class);
+        $this->adapter = $this->createMock(AdapterInterface::class);
+    }
+
+    /**
+     * Test delete media asset by id action
+     *
+     * @param int $mediaAssetId
+     * @dataProvider deleteMediaAssetTestScopes
+     */
+    public function testSuccessfulDeleteByIdExecution(int $mediaAssetId)
+    {
+        $this->getMediaAssetById->expects($this->at(0))
+            ->method('execute')
+            ->with($mediaAssetId)
+            ->willReturn($this->mediaAsset);
+
+        $this->resourceConnection->expects($this->at(0))
+            ->method('getConnection')
+            ->willReturn($this->adapter);
+
+        $tableName = 'media_gallery_asset';
+        $this->resourceConnection->expects($this->at(1))
+            ->method('getTableName')
+            ->with($tableName)
+            ->willReturn($tableName);
+
+        $this->mediaAsset->expects($this->once())
+            ->method('getId')
+            ->willReturn($mediaAssetId);
+
+        $this->adapter->expects($this->at(0))
+            ->method('delete')
+            ->with($tableName, 'id = ' . $mediaAssetId);
+
+        $this->deleteMediaAssetById->execute($mediaAssetId);
+    }
+
+    /**
+     * Assume that delete action will thrown an Exception
+     *
+     * @param int $mediaAssetId
+     * @dataProvider deleteMediaAssetTestScopes
+     */
+    public function testExceptionOnDeleteExecution(int $mediaAssetId)
+    {
+        $this->getMediaAssetById->expects($this->at(0))
```

Comment:
Please use `at` method only when you need to verify the sequence of calls (or when the same method is called several times with either different arguments or different return values)